pkg/sql/schemachanger/scgraph/graph.go

```diff
@@ -234,6 +263,18 @@ func (g *Graph) AddDepEdge(
 	return nil
 }
 
+// MarkOpEdgeAsNoOp marks an edge as no-op, so that no operations are emitted
+//during planning.
+func (g *Graph) MarkOpEdgeAsNoOp(edge *OpEdge) {
+	g.noOpEdges[edge] = true
+}
+
+// IsOpEdgeNoOp checks if an edge is marked as an edge that should emit no
+// operations.
+func (g *Graph) IsOpEdgeNoOp(edge *OpEdge) bool {
+	return g.noOpEdges[edge]
+}
```

Comment:
nit: can we change this terminology to say something about the edge being optimized away as opposed to being a `NoOp`?